lib/ios/BottomTabPresenter.m

```diff
@@ -1,68 +1,49 @@
 #import "BottomTabPresenter.h"
 #import "RNNTabBarItemCreator.h"
 #import "UIViewController+RNNOptions.h"
-#import "RNNDotIndicatorPresenter.h"
 #import "UIViewController+LayoutProtocol.h"
 
-@interface BottomTabPresenter ()
-@property(nonatomic, strong) RNNDotIndicatorPresenter* dotIndicatorPresenter;
-@end
-
 @implementation BottomTabPresenter
 
 - (instancetype)initWithDefaultOptions:(RNNNavigationOptions *)defaultOptions {
     self = [super init];
     self.defaultOptions = defaultOptions;
-    self.dotIndicatorPresenter = [[RNNDotIndicatorPresenter alloc] initWithDefaultOptions:defaultOptions];
     return self;
 }
 
-- (void)applyOptions:(RNNNavigationOptions *)options {
+- (void)applyOptions:(RNNNavigationOptions *)options child:(UIViewController *)child {
     RNNNavigationOptions * withDefault = [options withDefault:self.defaultOptions];
     
-    if (withDefault.bottomTab.badge.hasValue && [self.boundViewController.parentViewController isKindOfClass:[UITabBarController class]]) {
-        [self.boundViewController setTabBarItemBadge:withDefault.bottomTab.badge.get];
-    }
+    if (withDefault.bottomTab.badge.hasValue)
+        [child setTabBarItemBadge:withDefault.bottomTab.badge.get];
     
-    if (withDefault.bottomTab.badgeColor.hasValue && [self.boundViewController.parentViewController isKindOfClass:[UITabBarController class]]) {
-        [self.boundViewController setTabBarItemBadgeColor:withDefault.bottomTab.badgeColor.get];
-    }
+    if (withDefault.bottomTab.badgeColor.hasValue)
```

Comment:
Lets make this conditional a one-liner and remove empty lines around it. This way all lines where options are actually applied are grouped together and mistakes are easy to point out because each line should basically look like all others.